examples/nlp/language_modeling/conf/bert_pretraining_from_preprocessed_config.yaml

```diff
@@ -0,0 +1,72 @@
+# BERT Pretraining from Preprocessed (tokenized) data
+trainer:
+  gpus: 1 # the number of gpus, 0 for CPU, or list with gpu indices
+  num_nodes: 1
+  max_steps: 100 # precedence over max_epochs
+  num_sanity_val_steps: 0 # needed for bert pretraining from preproc
+  replace_sampler_ddp: false # needed for bert pretraining from preproc
+  accumulate_grad_batches: 1 # accumulates grads every k batches
+  precision: 16 # 16 to use AMP 
+  amp_level: O1 # O1 or O2 if using AMP 
+  distributed_backend: ddp
+  gradient_clip_val: 0.0
+  row_log_interval: 1
+  val_check_interval: 1.0 # check once per epoch .25 for 4 times per epoch
+  checkpoint_callback: false # provided by exp_manager
+  logger: false # provided by exp_manager
+
+model:
+  language_model:
+    pretrained_model_name: bert-base-uncased # huggingface model name
+    bert_checkpoint: null
+    bert_config:
```

Comment:
Most users want to use the default arch for training with their data or even train the default models. I think for these specific examples we should let huggingface to extract the default arch. Current setup makes switching between different BERT not easy as they need to pass the config file or change all the configs. How user can make the HF to extract the default configs? I mean how we can pass null for both bert_config_file and bert_config?

Reply:
agree with Vahid. This is also what i have mentioned in slack.

Reply:
I don't think you can do that with HF. If you know how to do that, then please let me know or point me to some documentation.